### PR TITLE
MessTenderFix ComplaintAPIsAdded MedicalAppointmentStatusAdded MedicalHistoryDateFieldAdded

### PR DESCRIPTION
**Pull request by architprasar**
Mess tender archived debug and fix
Upload image API
Medical side API refactor (status of appointment)
history creation date fix
@ashutoshc8101 Please review the request

### CODE REVIEW

File: api/authentication/models.py [hunk 1]
```diff
@@ -50,26 +54,30 @@ class User(AbstractBaseUser, PermissionsMixin):
...
 
     def __str__(self):
-      return self.email
+        return self.email
 
     class Meta:
         verbose_name = 'user'
         verbose_name_plural = 'users'
 
+
```
Comment (ashutoshc8101) [inline api/authentication/models.py:63]: ditto

File: api/authentication/models.py [hunk 2]
```diff
@@ -80,6 +88,7 @@ def is_authenticated(self):
     def __str__(self):
         return self.first_name + ' ' + self.last_name
 
+
```
Comment (ashutoshc8101) [inline api/authentication/models.py:91]: ditto

File: api/medical/models.py [hunk 3]
```diff
@@ -33,12 +33,19 @@ def __str__(self):
 
 
 class Appointment(models.Model):
+    STATUS = (
+        (1, 'in process'),
+        (2, 'accepted'),
+        (3, 'rejected'),
+
```
Comment (ashutoshc8101) [inline api/medical/models.py:40]: ditto

File: api/medical/views.py [hunk 4]
```diff
@@ -204,18 +204,18 @@ def get(self, request):
 
     def post(self, request):
         id = request.POST.get("id", None)
-        accepted = request.POST.get("accepted", None)
+        status = request.POST.get("status", None)
         appointment = Appointment.objects.filter(id=id).first()
-        if accepted == "true":
+        if status == 2:
```
Comment (ashutoshc8101) [inline api/medical/views.py:209]: USE a constant variable such as
```python
APPOINTMENT_STATUS = {
      'REJECTED': 0,
      'IN_PROGRES': 1,
      'APPROVED': 2
}

```

File: api/mess/urls.py [hunk 5]
```diff
@@ -40,6 +51,9 @@
          name="tender instance"),
     path('feedback/action/<int:pk>/',
          views.FeedbackActionView.as_view(), name="feedback action"),
+    path('complaint/action/<int:pk>/',
+         views.ComplaintActionView.as_view(), name="feedback action"),
+
```
Comment (ashutoshc8101) [inline api/mess/urls.py:56]: remove this empty line

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Minor comments.

File: api/medical/views.py (hunk 4 above)
Comment (architprasar) [reply api/medical/views.py:209]: the constant variable is used in the model this is only for the view

Comment (ashutoshc8101) [reply api/medical/views.py:209]: Share the same constant in both models.py and views.py

File: api/authentication/models.py [hunk 6]
```diff
@@ -50,26 +54,30 @@ class User(AbstractBaseUser, PermissionsMixin):
...
 class Staff(models.Model):
     # These fields tie to the roles!
 
     STAFF_ROLE_CHOICES = (
         (STAFF_ROLE_IDS['guard'], 'Guard'),
         (STAFF_ROLE_IDS['mess_manager'], 'Mess Manager'),
+        (STAFF_ROLE_IDS['medical_manager'], 'Medical Manager'),
+
```
[the remark this answers is not in the document]
Comment (architprasar) [reply api/authentication/models.py]: done

File: api/authentication/models.py [hunk 7]
```diff
@@ -14,10 +14,12 @@
 STAFF_ROLE_IDS = {
     'guard': 1,
     'mess_manager': 2,
-    'medical_manager':3,
+    'medical_manager': 3,
 }
 
 # Create your models here.
+
+
```
[the remark this answers is not in the document]
Comment (architprasar) [reply api/authentication/models.py]: done

File: api/authentication/models.py (hunk 1 above)
Comment (architprasar) [reply api/authentication/models.py:61]: done

File: api/authentication/models.py (hunk 2 above)
Comment (architprasar) [reply api/authentication/models.py]: done

**Comment by ashutoshc8101** [comment]
@architprasar Address all comments before asking for a review. If something has been done, reply with `done` to the review comment.

File: api/mess/urls.py (hunk 5 above)
Comment (architprasar) [reply api/mess/urls.py]: done

**Comment by architprasar** [comment]
@ashutoshc8101 please review

File: api/medical/views.py (hunk 4 above)
Comment (architprasar) [reply api/medical/views.py]: done

**Review by ashutoshc8101 — APPROVED** [review]

**Pull request merged**